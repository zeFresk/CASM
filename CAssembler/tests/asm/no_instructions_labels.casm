.at 20
a: .word 042
ct0: .word 000
ret: .word 000
ret_ptr: .word ret

.at 30 ; function
add: ; function which takes one argument
lda ct0
jaz add_start
return_adr: .word 000
parameter: .word 000
return_ptr: .word 000
add_start:
lda parameter
add a
add_ret:
sti return_ptr
lda ct0
jaz return_adr

.start 60
lda ret_here
sta [ add + 2 ]
lda a
sta [ add + 3 ]
lda ret_ptr
sta [ add + 4 ]
ret_here: hrs 00
