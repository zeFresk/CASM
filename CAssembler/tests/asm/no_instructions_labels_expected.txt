60
20: 042
21: 000
22: 000
23: 022
30: 221
31: 835
32: 000
33: 000
34: 000
35: 233
36: 620
37: 534
38: 221
39: 832
60: 266
61: 432
62: 220
63: 433
64: 223
65: 434
66: 900
